### ISSUE TITLE
Unable to download attachments on Android 10, Librus.

### ISSUE THREAD

**Issue by gmalinowski** [szkolny-eu/szkolny-android#185]
While trying to download the attachment from inbox messages, the following error is occurring:

```
Kod błędu: 50 (on LibrusSandboxDownloadAttachment) /storage/emulated/0/Download/Szkolny.eu/FILE_NAME:
open failed: ENOENT (No such file or directory)
```

**Comment by kuba2k2** [comment]
Does the filename contain any special characters? Does the mentioned directory (not the file, the parent directory) exist?

**Comment by gmalinowski** [comment]
There are several attachments I think all without special characters like '03.mp3', none of them work. I forgot to add the error title which is *Błąd odpowiedzi serwera. Dziennik może być przeciążony lub mieć przerwę techniczną.*

And yes, I tried to create the parent directory, it didn't help but the err msg has changed.

![signal-2023-10-26-141849](https://github.com/szkolny-eu/szkolny-android/assets/26151125/3ae5cab5-5084-4a23-b0fe-d9c42028a4ab)

![signal-2023-10-26-141832](https://github.com/szkolny-eu/szkolny-android/assets/26151125/27b12060-3b4f-4c3a-84af-1d00ab8b83d2)

**Comment by kuba2k2** [comment]
Well, then, did you grant storage permissions to the app? Are you using the latest version?

The error title is a generic one, happens for all kinds of exceptions.

**Comment by gmalinowski** [comment]
Yes, I've the latest version (4.13.6) and of course, I granted the required permissions.

**Comment by gmalinowski** [comment]
Wow I've solved the problem, kinda. I don't know much about android apps but this is wired. Actually, it has logic in it, the app cannot fetch data from external sources, i guess.

In the advanced settings in app info, it requires to grant permissions to "Install unknown apps" / "Allow from this source". The app was installed from the apk not from the google play. Like on the screenshot below.

![signal-2023-10-26-144215](https://github.com/szkolny-eu/szkolny-android/assets/26151125/36983049-ffe0-4be2-979d-ff3bc8b7d838)

Right now, it's downloading the files, but it's still throwing an error:
![signal-2023-10-26-144246](https://github.com/szkolny-eu/szkolny-android/assets/26151125/4a40ec24-fd0c-4689-a3bb-9ea4f8678904)
